file.go

```diff
@@ -1,37 +1,174 @@
 package jcsv
 
-import "os"
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+)
 
 type file struct {
-	// TODO: define variables if needed
+	Data map[string]interface{}
 }
 
 func ParseJsonFile(path string) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	//open JSON file
+	jsonFile, err := os.Open(path)
+
+	//if an error occurred
+	if err != nil {
+		fmt.Println(err)
+		return file{}, err
+	}
+
+	//defer the closing of file till the function ends
+	defer jsonFile.Close()
+
+	//read complete json file and store in byteValue
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	//if an error occurred while reading JSON file
+
+	if err != nil {
+		fmt.Println(err)
+		return file{}, err
+	}
+
+	//create file object
+	var myFile file
+	//convert JSON read from file in byteValue to "Data" inside file stucture
+	json.Unmarshal(byteValue, &myFile.Data)
+
+	return myFile, err
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+	//create file object
+	var myFile file
+	//read complete json file and store in byteValue
+	byteValue, err := ioutil.ReadAll(f)
+
+	//if an error occurred while reading JSON file
+	if err != nil {
+		fmt.Println(err)
+		return file{}, nil
+	}
+	//convert JSON read from file in byteValue to "Data" inside file stucture
+
+	json.Unmarshal(byteValue, &myFile)
+
+	return myFile, err
 }
 
 func ParseCsvFile(path string, hasHeaders bool) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	//open CSV file
+	csvFile, err := os.Open(path)
+
+	//if an error occurred
+	if err != nil {
+		return file{}, err
+	}
+
+	//defer closing of file
+	defer csvFile.Close()
+
+	//get CSV Reader instance
+	csvReader := csv.NewReader(csvFile)
+	//create an object to hold csv data
+	var CSVData [][]string
+	//read all data into CSVData
+	CSVData, err = csvReader.ReadAll()
+
+	//if an error occurred
+	if err != nil {
+		return file{}, nil
+	}
+
+	//create file object
+	var myFile file
+	//make map
+	myFile.Data = make(map[string]interface{})
+	//store csv data into file structure
+	i := 0
+	if !hasHeaders {
+		i = 1
+	}
+	for ; i < len(CSVData); i = i + 1 {
+		myFile.Data["Record "+strconv.Itoa(i)] = CSVData[i]
```

Comment:
You won't need to name each record if you use an array, i.e. `[]map[string]interface{}`.

Reply:
all right